CorsixTH/Lua/humanoid_actions/seek_reception.lua

```diff
@@ -78,13 +79,16 @@ local function action_seek_reception_start(action, humanoid)
       local orientation = desk.object_type.orientations[desk.direction]
       local x = desk.tile_x + orientation.use_position[1]
       local y = desk.tile_y + orientation.use_position[2]
-      local this_score = humanoid.world:getPathDistance(humanoid.tile_x, humanoid.tile_y, x, y)
 
-      this_score = this_score + desk:getUsageScore()
-      if not score or this_score < score then
-        -- It is better, or the first one!
-        score = this_score
-        best_desk = desk
+      local distance_from_desk = humanoid.world:getPathDistance(humanoid.tile_x, humanoid.tile_y, x, y)
+      if distance_from_desk ~= false then
```

Comment:
Just `if distance_from_desk then` (suggested code changes seem to have disappeared). `0` evaluates to `true` in Lua.

Reply:
The 0 evaluation to true is very weird coming from Python, thanks for the reminder !

```suggestion
      if distance_from_desk then
```

Reply:
I agree it's weird at first sight, but in a sense it's cleaner. In Lua `true` and `false` are boolean values, numbers are values exclusively used for computing numeric results.

The reason why `0/1` are often interpreted as boolean comes from original C, which didn't have booleans by itself and instead borrowed the CPU-level notion of it, a single bit that is either `0` or `1`.

While pretty soon everybody wrote
``` C
#define TRUE 1
#define FALSE 0
```
in C to have names for boolean values, the `0/1` idea was still copied to many other languages at first, and then kept when they added proper `true/false` notions for backwards-compatibility.